lib/minitest/speed.rb

```diff
@@ -24,37 +24,75 @@ module Minitest::Speed
 
   @@max_teardown_time = 1.0
 
+  ##
+  # Default way of getting the current time.
+  # @@clock_time_method = proc { Minitest.clock_time }
+
+  mc = class << self; self; end
+
+  ##
+  # Default way of getting the current time.
+
+  mc.attr_accessor :clock_time_method
+
+  self.clock_time_method = -> { Minitest.clock_time }
+
+
   def before_setup # :nodoc:
     super
 
-    @setup_t0 = Minitest.clock_time
+    @setup_t0 = Minitest::Speed.clock_time_method.call
   end
 
   def after_setup # :nodoc:
-    delta = Minitest.clock_time - @setup_t0
+    delta = Minitest::Speed.clock_time_method.call - @setup_t0
 
-    @test_t0 = Minitest.clock_time
+    @test_t0 = Minitest::Speed.clock_time_method.call
 
-    assert_operator delta, :<=, @@max_setup_time, "max_setup_time exceeded"
+    assert_operator delta, :<=, @@max_setup_time, "max_setup_time exceeded" unless @permit_slow_setup
+    @permit_slow_setup = false
```

Comment:
I don't understand these lines. Why are you resetting? These are on the test instance.